Since GNU/Hurd doesn't support byte-range file locks, shamelessly
disable them to allow the SMB client library to work (it uses TDB
to access /etc/samba/private/secrets.tdb, for instance.)

--- samba/lib/tdb/common/lock.c	2010-02-08 16:12:57.000000000 +0100
+++ samba/lib/tdb/common/lock.c	2012-03-01 23:39:02.000000000 +0100
@@ -48,7 +48,7 @@ int tdb_brlock(struct tdb_context *tdb,
 	struct flock fl;
 	int ret;
 
-	if (tdb->flags & TDB_NOLOCK) {
+	if (1) {
 		return 0;
 	}
